csrc/runtime/executor.cpp

```diff
@@ -244,6 +244,50 @@ std::vector<at::Tensor> ExprEvalExecutor::run(
   return outputs;
 }
 
+bool KernelExecutor::supported(Fusion* fusion) {
+  FUSER_PERF_SCOPE("KernelExecutor::supported");
+
+  auto has_only_scalar_inputs = [](Expr* expr) -> bool {
```

Comment:
Since this function might eventually do stuff other than verify scalar tensors, it might make sense to put most of this implementation into a static function  like `isPureCPUScalarFusion(fusion)`.

Reply:
Yes, I agree.
I am still playing around with some other configurations of inputs to make sure these are the only rules.
I'll finalize the PR after that.